homeassistant/components/cover/opengarage.py

```diff
@@ -0,0 +1,222 @@
+ATTR_DOOR_STATE = "door_state"
+
+STATE_OPENING = "opening"
+STATE_CLOSING = "closing"
+STATE_STOPPED = "stopped"
+STATE_OFFLINE = "offline"
+
+STATES_MAP = {
+    0: STATE_CLOSED,
+    1: STATE_OPEN
+}
+
+
+# Validation of the user's configuration
+COVER_SCHEMA = vol.Schema({
+    vol.Required(CONF_DEVICEKEY): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_NAME): cv.string
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_COVERS): vol.Schema({cv.slug: COVER_SCHEMA}),
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup OpenGarage covers."""
+    covers = []
+    devices = config.get(CONF_COVERS)
+
+    for device_id, device_config in devices.items():
+        args = {
+            CONF_NAME: device_config.get(CONF_NAME),
+            CONF_HOST: device_config.get(CONF_HOST),
+            CONF_PORT: device_config.get(CONF_PORT),
+            "device_id": device_config.get(CONF_DEVICE, device_id),
+            CONF_DEVICEKEY: device_config.get(CONF_DEVICEKEY)
+        }
+
+        covers.append(OpenGarageCover(hass, args))
+
+    add_devices(covers)
+
+
+class OpenGarageCover(CoverDevice):
+    """Representation of a OpenGarage cover."""
+
+    # pylint: disable=no-self-use
+    def __init__(self, hass, args):
+        """Initialize the cover."""
+        self.opengarage_url = 'http://{}:{}'.format(
+            args[CONF_HOST],
+            args[CONF_PORT])
+        self.hass = hass
+        self._name = args[CONF_NAME]
+        self.device_id = args['device_id']
+        self._devicekey = args[CONF_DEVICEKEY]
+        self._state = STATE_UNKNOWN
+        self._state_before_move = None
+        self.dist = None
+        self.signal = None
+        self._available = True
+
+        # Lets try to get the configured name if not provided.
+        try:
+            if self._name is None:
+                doorconfig = self._get_status()
+                if doorconfig["name"] is not None:
+                    self._name = doorconfig["name"]
+            self.update()
+        except requests.exceptions.ConnectionError as ex:
+            _LOGGER.error('Unable to connect to server: %(reason)s',
+                          dict(reason=ex))
+            self._state = STATE_OFFLINE
+            self._available = False
+            self._name = DEFAULT_NAME
+        except KeyError as ex:
+            _LOGGER.warning('OpenGarage device %(device)s seems to be offline',
+                            dict(device=self.device_id))
+            self._name = DEFAULT_NAME
+            self._state = STATE_OFFLINE
+            self._available = False
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """No polling needed for a demo cover."""
```

Comment:
Stale comment. The default for polling is already `True` so you can just remove this whole property.